subprojects/files/src/main/java/org/gradle/internal/file/FileMetadataSnapshot.java

```diff
@@ -31,4 +31,23 @@
      * Note: always 0 for directories and missing files.
      */
     long getLength();
+
+    /**
+     * How the file with the metadata was accessed.
+     */
+    AccessType getAccessType();
```

Comment:
We should clarify here how "transitive symlinking" works, i.e. that this will be `VIA_SYMLINK` when any ancestor is a symlink (right?).

Reply:
It won't be `VIA_SYMLINK` if the ancestor is a symlink. This will only be via symlink if the queried path is a symlink. Transitive here should mean that if the queried path is a symlink pointing to a symlink pointing to a file, then the access type is `VIA_SYMLINK` and the snapshot is about the metadata of the target of the second symlink.

Reply:
Can you please add this as Javadoc?

Reply:
Added.